### PR TITLE
expression: fix buildin function found_rows()

### LINKED ISSUE #3131: function found_rows() return wrong value
after select statement, select found_row() return wrong value. for example:
(1) select * from <table_name> // return n rows, n > 0
(2) select found_rows(); // should return n, but 0

### PR DESCRIPTION
**Pull request by ueizhou**
This PR fix #3131. fix buildin function found_rows().
But not support `SQL_CALC_FOUND_ROWS` yet, it should be implemented in another PR.
`server/conn.go: invoke AddFoundRows()` after query statement  succeed.
`sessionctx/variable/session.go`: add variable `LastFoundRows` to hold last found rows. Using `LastFoundRows` not `LastStmtCtx.foundRows` because `StmtCtx` holds `foundRows` and affectRows both.
`expression/buildin_info.go`: use LastFoundRows

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -149,6 +149,9 @@ type SessionVars struct {
 	// GlobalAccessor is used to set and get global variables.
 	GlobalVarsAccessor GlobalVarAccessor
 
+	// LastFoundRows is the last found rows of previous query statement
```
Comment (zhexuany) [inline sessionctx/variable/session.go]: Maybe LastFoundRow is better than LastFoundRows. It is just a unit64 not a []uint64.

Comment (ueizhou) [inline sessionctx/variable/session.go]: Just keep same as function name :)

Comment (zhexuany) [inline sessionctx/variable/session.go]: @hanfei1991 what do you think？ keep same as function name or remove the "s"

Comment (hanfei1991) [inline sessionctx/variable/session.go]: I think it's better to remove 's'

Comment (ueizhou) [inline sessionctx/variable/session.go]: Sad. FoundRows is clear and no ambiguity in mysql world. If remove 's', all related members and functions of the StatementContext struct should be modified. (sessionctx/variable/session.go:260)

Comment (coocood) [inline sessionctx/variable/session.go]: `FoundRow` should return a row instead of an integer.
`FoundRows` can be interpreted as `found number of rows`.

Comment (zhexuany) [inline sessionctx/variable/session.go]: @ueizhou In that case, considering to add a comment for clarifying this. Without any pre-knowledge, I will assume `LastRows` is an array.

Comment (ueizhou) [inline sessionctx/variable/session.go]: @zhexuany Good suggestion! it's done.

**Comment by ueizhou** [comment]
PTAL @zhexuany @coocood @hanfei1991 @shenli

**Comment by coocood** [comment]
@ueizhou 
Any tests to cover the bug?

**Comment by ueizhou** [comment]
The return value of this buildin function lies on result of previous query statement. Where can I found any example?

**Comment by coocood** [comment]
@ueizhou 
According to MySQL document 
https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_found-rows
found rows is useful only when used with`SQL_CALC_FOUND_ROWS` and `LIMIT`.

But without `SQL_CALC_FOUND_ROWS1`, the result should be the number of rows returned by the last query.

Your implementation update found rows in the `server` package which is not the best place to do it.

If you implement it in `executor` package, `recordSet` type, it would be much easier to write test.
There are many test cases in the package you can reference.

**Comment by ueizhou** [comment]
@coocood I moved the code to package executor and add test cases. But not implement  SQL_CALC_FOUND_ROWS and LIMIT in this PR. I think it's should be implemented in another PR.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@ueizhou Thanks!

**Pull request merged**